### PR TITLE
Finish story 26

### PR DESCRIPTION
**Pull request by meghanstovall**
Finish story 26

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
   def new
-
   end
 
   def show
     @order = Order.find(params[:id])
   end
 
   def create
-    order = Order.create(order_params)
```
Comment (meghanstovall) [inline app/controllers/orders_controller.rb:12]: Add a user because an order cant be created without one

**Comment by palworth** [comment]
User story 26 complete and ready for master.

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -19,13 +19,21 @@ def create
           })
       end
       session.delete(:cart)
-      redirect_to "/orders/#{order.id}"
+      flash[:success] = "Your order has been placed!"
```
Comment (meghanstovall) [inline app/controllers/orders_controller.rb:22]: Add conditional for redirection depending on what type of user they are

Comment (meghanstovall) [inline app/controllers/orders_controller.rb:22]: Add flash message once order has been placed

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -19,13 +19,21 @@ def create
...
+      else
+        redirect_to "/profile/orders"
+      end
     else
       flash[:notice] = "Please complete address form to create an order."
       render :new
     end
   end
 
+  def index
```
Comment (meghanstovall) [inline app/controllers/orders_controller.rb:34]: Add index action for orders

File: app/models/order.rb [hunk 4]
```diff
@@ -1,10 +1,20 @@
-class Order <ApplicationRecord
+class Order < ApplicationRecord
   validates_presence_of :name, :address, :city, :state, :zip
 
+  belongs_to :user
```
Comment (meghanstovall) [inline app/models/order.rb:4]: make an order belong to a user

File: app/models/order.rb [hunk 5]
```diff
@@ -1,10 +1,20 @@
...
   has_many :item_orders
   has_many :items, through: :item_orders
+  before_save :default_values
+
```
Comment (meghanstovall) [inline app/models/order.rb:8]: Add before_save to set the orders default status to pending

File: app/models/order.rb [hunk 6]
```diff
@@ -1,10 +1,20 @@
...
+    self.status = 'pending'
+  end
 
   def grandtotal
     item_orders.sum('price * quantity')
   end
+
+  def total_item_quantity
+    item_orders.sum('quantity')
+  end
```
Comment (meghanstovall) [inline app/models/order.rb:19]: Add model method that sums up the item_orders quantity

File: app/models/user.rb [hunk 7]
```diff
@@ -11,6 +11,8 @@ class User < ApplicationRecord
 
   validates_uniqueness_of :email
 
+  has_many :orders
+
```
Comment (meghanstovall) [inline app/models/user.rb:15]: Made users have many orders

File: app/views/orders/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<h2> My Orders </h2>
```
Comment (meghanstovall) [inline app/views/orders/index.html.erb:1]: Add index view for orders

File: config/routes.rb [hunk 9]
```diff
@@ -47,6 +47,7 @@
   patch "/profile/:profile_id", to: 'users#update_password'
 
   patch '/profile', to: 'users#update'
+  get '/profile/orders', to: 'orders#index'
   get '/login', to: 'sessions#new'
```
Comment (meghanstovall) [inline config/routes.rb:51]: Added a route for the orders index action

File: db/migrate/20200222171528_add_user_to_orders.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+class AddUserToOrders < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :orders, :user, foreign_key: true
+  end
```
Comment (meghanstovall) [inline db/migrate/20200222171528_add_user_to_orders.rb:4]: Add a foreign key of the user id to the orders table

File: db/migrate/20200222172854_add_status_to_orders.rb [hunk 11]
```diff
@@ -0,0 +1,6 @@
+class AddStatusToOrders < ActiveRecord::Migration[5.1]
+  def change
+    add_column :orders, :status, :string, default: 'pending'
+    Order.all.where(status: nil).update_all(status: 'pending')
+  end
```
Comment (meghanstovall) [inline db/migrate/20200222172854_add_status_to_orders.rb:5]: Made status default value of 'pending'

File: db/schema.rb [hunk 12]
```diff
@@ -57,6 +57,9 @@
     t.integer "zip"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.bigint "user_id"
+    t.string "status"
+    t.index ["user_id"], name: "index_orders_on_user_id"
   end
```
Comment (meghanstovall) [inline db/schema.rb:63]: Add users and status column to orders table

File: db/schema.rb [hunk 13]
```diff
@@ -83,5 +86,6 @@
   add_foreign_key "item_orders", "items"
   add_foreign_key "item_orders", "orders"
   add_foreign_key "items", "merchants"
+  add_foreign_key "orders", "users"
```
Comment (meghanstovall) [inline db/schema.rb:89]: Add user to orders

File: spec/features/cart/checkout_spec.rb [hunk 14]
```diff
@@ -27,6 +27,43 @@
 
       expect(current_path).to eq("/orders/new")
     end
+
```
Comment (meghanstovall) [inline spec/features/cart/checkout_spec.rb:30]: Test that cart is empty after order is placed and a flash message tells the user the order was placed

File: spec/features/items/destroy_spec.rb [hunk 15]
```diff
@@ -27,11 +27,21 @@
...
     end
 
-    it 'I can not delete items with orders' do 
+    it 'I can not delete items with orders' do
       bike_shop = Merchant.create(name: "Brian's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
       chain = bike_shop.items.create(name: "Chain", description: "It'll never break!", price: 50, image: "https://www.rei.com/media/b61d1379-ec0e-4760-9247-57ef971af0ad?size=784x588", inventory: 5)
       review_1 = chain.reviews.create(title: "Great place!", content: "They have great bike stuff and I'd recommend them to anyone.", rating: 5)
-      order_1 = Order.create!(name: 'Meg', address: '123 Stang St', city: 'Hershey', state: 'PA', zip: 80218)
+
+      tim = User.create(name: 'Tim',
```
Comment (meghanstovall) [inline spec/features/items/destroy_spec.rb:35]: Add a user for the orders

File: spec/features/items/index_spec.rb [hunk 16]
```diff
@@ -104,8 +104,8 @@
     end
 
     it "can see stats for top 5 most popular and least 5 popular" do
-      order_1 = Order.create(name: "Meg", address: "123 Turing St", city: "Denver", state: "CO", zip: "80020")
-      order_2 = Order.create(name: "Mike", address: "123 Turing St", city: "Denver", state: "CO", zip: "80020")
+      order_1 = @user.orders.create(name: "Meg", address: "123 Turing St", city: "Denver", state: "CO", zip: "80020")
```
Comment (meghanstovall) [inline spec/features/items/index_spec.rb:107]: created the orders with a user

File: spec/features/merchants/destroy_spec.rb [hunk 17]
```diff
@@ -48,25 +48,26 @@
...
-
-      fill_in :name, with: name
-      fill_in :address, with: address
-      fill_in :city, with: city
-      fill_in :state, with: state
-      fill_in :zip, with: zip
+      user = User.create(name: "Bert",
+                        street_address: "123 Sesame St.",
+                        city: "NYC",
+                        state: "New York",
```
Comment (meghanstovall) [inline spec/features/merchants/destroy_spec.rb:54]: Add a user for the order

File: spec/features/merchants/destroy_spec.rb [hunk 18]
```diff
@@ -48,25 +48,26 @@
...
+      fill_in :state, with: user.state
+      fill_in :zip, with: user.zip
 
       click_button "Create Order"
 
       visit "/merchants/#{meg.id}"
       expect(page).to_not have_link("Delete Merchant")
-
-      # visit "/merchants/#{brian.id}"
-      # expect(page).to have_link("Delete Merchant")
```
Comment (meghanstovall) [inline spec/features/merchants/destroy_spec.rb:69]: delete unnecessary comments

File: spec/features/merchants/statistics_spec.rb [hunk 19]
```diff
@@ -16,9 +16,18 @@
       @pull_toy = @brian.items.create(name: "Pull Toy", description: "Great pull toy!", price: 10, image: "http://lovencaretoys.com/image/cache/dog/tug-toy-dog-pull-9010_2-800x800.jpg", inventory: 32)
       @dog_bone = @brian.items.create(name: "Dog Bone", description: "They'll love it!", price: 20, image: "https://img.chewy.com/is/image/catalog/54226_MAIN._AC_SL1500_V1534449573_.jpg", active?:false, inventory: 21)
 
-      @order_1 = Order.create!(name: 'Meg', address: '123 Stang Ave', city: 'Hershey', state: 'PA', zip: 17033)
-      @order_2 = Order.create!(name: 'Brian', address: '123 Zanti St', city: 'Denver', state: 'CO', zip: 80204)
-      @order_3 = Order.create!(name: 'Mike', address: '123 Dao St', city: 'Denver', state: 'CO', zip: 80210)
+      @tim = User.create(name: 'Tim',
+                      street_address: '123 Turing St',
```
Comment (meghanstovall) [inline spec/features/merchants/statistics_spec.rb:20]: Add a user so the orders can be created

File: spec/features/orders/creation_spec.rb [hunk 20]
```diff
@@ -1,16 +1,10 @@
-# When I fill out all information on the new order page
-# And click on 'Create Order'
-# An order is created and saved in the database
-# And I am redirected to that order's show page with the following information:
-#
-# - Details of the order:
-
```
Comment (meghanstovall) [inline spec/features/orders/creation_spec.rb:7]: delete unnecessary comments and add require rails_helper

File: spec/features/orders/creation_spec.rb [hunk 21]
```diff
@@ -29,17 +23,21 @@
...
-      zip = 10001
-
-      fill_in :name, with: name
-      fill_in :address, with: address
-      fill_in :city, with: city
-      fill_in :state, with: state
-      fill_in :zip, with: zip
+      user = User.create(name: "Bert",
+                        street_address: "123 Sesame St.",
+                        city: "NYC",
```
Comment (meghanstovall) [inline spec/features/orders/creation_spec.rb:28]: Add a user so the order could be made

File: spec/features/orders/creation_spec.rb [hunk 22]
```diff
@@ -48,11 +46,11 @@
       expect(current_path).to eq("/orders/#{new_order.id}")
 
       within '.shipping-address' do
-        expect(page).to have_content(name)
-        expect(page).to have_content(address)
-        expect(page).to have_content(city)
-        expect(page).to have_content(state)
-        expect(page).to have_content(zip)
+        expect(page).to have_content(new_order.name)
```
Comment (meghanstovall) [inline spec/features/orders/creation_spec.rb:49]: change test to make sure the right order is on the page

File: spec/features/orders/creation_spec.rb [hunk 23]
```diff
@@ -89,24 +87,26 @@
...
-      city = "NYC"
-      state = "New York"
-      zip = 10001
-
-      fill_in :name, with: name
-      fill_in :address, with: address
-      fill_in :city, with: city
-      fill_in :state, with: state
-      fill_in :zip, with: zip
+      user = User.create!(name: "Bert",
```
Comment (meghanstovall) [inline spec/features/orders/creation_spec.rb:90]: Add a user so the order could be made

File: spec/models/item_orders_spec.rb [hunk 24]
```diff
@@ -17,7 +17,16 @@
     it 'subtotal' do
       meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
       tire = meg.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
-      order_1 = Order.create!(name: 'Meg', address: '123 Stang Ave', city: 'Hershey', state: 'PA', zip: 17033)
+      tim = User.create(name: 'Tim',
```
Comment (meghanstovall) [inline spec/models/item_orders_spec.rb:20]: Add user for the order

File: spec/models/item_spec.rb [hunk 25]
```diff
@@ -43,7 +43,16 @@
 
     it 'no orders' do
       expect(@chain.no_orders?).to eq(true)
-      order = Order.create(name: 'Meg', address: '123 Stang Ave', city: 'Hershey', state: 'PA', zip: 17033)
+      tim = User.create(name: 'Tim',
```
Comment (meghanstovall) [inline spec/models/item_spec.rb:46]: Add user for the order

File: spec/models/item_spec.rb [hunk 26]
```diff
@@ -61,8 +70,17 @@
       @dog_bed = @brian.items.create(name: "Dog Bed", description: "They'll love it!", price: 15, image: "https://img.chewy.com/is/image/catalog/54226_MAIN._AC_SL1500_V1534449573_.jpg", active?:false, inventory: 21)
       @dog_dish = @brian.items.create(name: "Dog Dish", description: "They'll love it!", price: 5, image: "https://img.chewy.com/is/image/catalog/54226_MAIN._AC_SL1500_V1534449573_.jpg", active?:false, inventory: 21)
 
-      @order_1 = Order.create(name: "Meg", address: "123 Turing St", city: "Denver", state: "CO", zip: "80020")
-      @order_2 = Order.create(name: "Mike", address: "123 Turing St", city: "Denver", state: "CO", zip: "80020")
+      @tim = User.create(name: 'Tim',
```
Comment (meghanstovall) [inline spec/models/item_spec.rb:73]: Add user so the orders could be created

File: spec/models/merchant_spec.rb [hunk 27]
```diff
@@ -21,7 +21,16 @@
     it 'no_orders' do
       expect(@meg.no_orders?).to eq(true)
 
-      order_1 = Order.create!(name: 'Meg', address: '123 Stang Ave', city: 'Hershey', state: 'PA', zip: 17033)
+      tim = User.create(name: 'Tim',
```
Comment (meghanstovall) [inline spec/models/merchant_spec.rb:24]: Add user so the user could be created

File: spec/models/merchant_spec.rb [hunk 28]
```diff
@@ -41,9 +50,18 @@
 
     it 'distinct_cities' do
       chain = @meg.items.create(name: "Chain", description: "It'll never break!", price: 40, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 22)
-      order_1 = Order.create!(name: 'Meg', address: '123 Stang Ave', city: 'Hershey', state: 'PA', zip: 17033)
-      order_2 = Order.create!(name: 'Brian', address: '123 Brian Ave', city: 'Denver', state: 'CO', zip: 17033)
-      order_3 = Order.create!(name: 'Dao', address: '123 Mike Ave', city: 'Denver', state: 'CO', zip: 17033)
+      tim = User.create(name: 'Tim',
+                      street_address: '123 Turing St',
+                      city: 'Denver',
```
Comment (meghanstovall) [inline spec/models/merchant_spec.rb:55]: Add a user for the orders

File: spec/models/order_spec.rb [hunk 29]
```diff
@@ -10,6 +10,7 @@
   end
 
   describe "relationships" do
+    it {should belong_to :user}
```
Comment (meghanstovall) [inline spec/models/order_spec.rb:13]: test that orders belong to a user

File: spec/models/order_spec.rb [hunk 30]
```diff
@@ -18,11 +19,12 @@
     before :each do
       @meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
       @brian = Merchant.create(name: "Brian's Dog Shop", address: '125 Doggo St.', city: 'Denver', state: 'CO', zip: 80210)
+      @mike = User.create(name: "Mike", street_address: '125 Turing St.', city: 'Denver', state: 'CO', zip: 80210, email: 'mike@gmail.com', password: 'password1', password_confirmation: 'password1')
```
Comment (meghanstovall) [inline spec/models/order_spec.rb:23]: Add user so orders could be created

File: spec/models/user_spec.rb [hunk 31]
```diff
@@ -15,6 +15,10 @@
 		it {should validate_uniqueness_of :email}
 	end
 
+	describe 'relationships' do
+		it {should have_many :orders}
+	end
```
Comment (meghanstovall) [inline spec/models/user_spec.rb:20]: test that items have many orders

**Pull request merged**